libs/SmartSync/src/com/salesforce/androidsdk/smartsync/util/SyncTarget.java

```diff
@@ -26,110 +26,19 @@
-
-import org.json.JSONArray;
 import org.json.JSONException;
 import org.json.JSONObject;
 
-import com.salesforce.androidsdk.smartsync.manager.SyncManager;
-
 /**
- * Target for sync i.e. set of objects to download from server
+ * Target for sync down:
+ * - what records to download from server
+ * - how to download those records
  */
-public abstract class SyncTarget {
-
-    // Constants
-	public static final String QUERY_TYPE = "type";
-    public static final String ANDROID_IMPL = "androidImpl";
-
-    // Fields
-	protected QueryType queryType;
-    protected int totalSize; // set during a fetch
-
-	/**
-	 * Build SyncTarget from json
-	 * @param target as json
-	 * @return
-	 * @throws JSONException 
-	 */
-	@SuppressWarnings("unchecked")
-	public static SyncTarget fromJSON(JSONObject target) throws JSONException {
-		if (target == null)
-			return null;
-
-		QueryType queryType = QueryType.valueOf(target.getString(QUERY_TYPE));
-
-        switch (queryType) {
-        case mru:     return MruSyncTarget.fromJSON(target);
-        case sosl:    return SoslSyncTarget.fromJSON(target);
-        case soql:    return SoqlSyncTarget.fromJSON(target);
-        case custom:
-        default:
-            try {
-                Class<? extends SyncTarget> implClass = (Class<? extends SyncTarget>) Class.forName(target.getString(ANDROID_IMPL));
-                Method method = implClass.getMethod("fromJSON", JSONObject.class);
-                return (SyncTarget) method.invoke(null, target);
-            } catch (Exception e) {
-                throw new RuntimeException(e);
-            }
-        }
-	}
-
-    /**
-     * @return query type of target
-     */
-
-	/**
-	 * @return json representation of target
-	 * @throws JSONException
-	 */
-	public abstract JSONObject asJSON() throws JSONException;
-
-    /**
-     * @param maxTimeStamp
-     * @return next record fetched
-     */
-
-    /**
-     * Start fetching records conforming to target
-     * If a value for maxTimeStamp greater than 0 is passed in, only records created/modified after maxTimeStamp should be returned
-     * @param syncManager
-     * @param maxTimeStamp
-     * @throws IOException, JSONException
-     */
-    public abstract JSONArray startFetch(SyncManager syncManager, long maxTimeStamp) throws IOException, JSONException;
-
-    /**
-     * Continue fetching records conforming to target if any
-     * @param syncManager
-     * @return null if there are no more records to fetch
-     * @throws IOException, JSONException
-     */
-    public abstract JSONArray continueFetch(SyncManager syncManager) throws IOException, JSONException;
-
-    /**
-     * @return number of records expected to be fetched - is set when startFetch() is called
-     */
-    public int getTotalSize() {
-        return totalSize;
-    }
-
-    /**
-     * @return QueryType of this target
-     */
-    public QueryType getQueryType() {
-        return queryType;
-    }
+public interface SyncTarget {
```

Comment:
Seems like a good idea.  I thought the same for iOS, but didn't get there on the first pass.